### PR TITLE
test: move check_node_connections to util

### PR DESCRIPTION
**Pull request by brunoerg**
This function (`check_node_connections`) exists in `feature_anchors.py` and `p2p_add_connections.py` and does the same thing in both, this PR moves this function to util to avoid repetition and facilitate maintenance.

### CODE REVIEW

**Review by lsilva01 — APPROVED** [review]
tACK https://github.com/bitcoin/bitcoin/pull/23392/commits/5ce601c32791a4e800b633407b4c5f44f52c6c8f on Ubuntu 20.04

**Review by stratospher — COMMENTED** [review]
ACK ab9bfc8.

File: test/functional/test_framework/util.py [hunk 1]
```diff
@@ -445,6 +446,14 @@ def set_node_times(nodes, t):
         node.setmocktime(t)
 
 
+def check_node_connections(*, node, num_in, num_out):
+    info = node.getnetworkinfo()
+    if info["connections_in"] != num_in:
+        raise AssertionError("connections_in: %s != %s" % (str(info["connections_in"]), str(num_in)))
+    if info["connections_out"] != num_out:
+        raise AssertionError("connections_out: %s != %s" % (str(info["connections_out"]), str(num_out)))
```
Comment (MarcoFalke) [inline test/functional/test_framework/util.py:454]: why is this changed?

**Review by maflcko — COMMENTED** [review]
I think all commits can be squashed into one to allow for easy `--color-moved=dimmed-zebra`.

**Comment by brunoerg** [comment]
> I think all commits can be squashed into one to allow for easy `--color-moved=dimmed-zebra`.

Done.

File: test/functional/test_framework/util.py (hunk 1 above)
Comment (brunoerg) [reply test/functional/test_framework/util.py:454]: To get more friendly messages.

Comment (brunoerg) [reply test/functional/test_framework/util.py:454]: The difference is minimal, example: `AssertionError: connections_in: 5 != 3`, with `assert_equal`: `AssertionError: not(5 == 4)`. I can change back to assert_equal if you think its better, np.

Comment (brunoerg) [reply test/functional/test_framework/util.py:454]: Yes, considering the stack trace, there is no difference. Going to change back to assert_equal.

Comment (brunoerg) [reply test/functional/test_framework/util.py:454]: Done.

**Comment by laanwj** [comment]
Code review ACK 988024fe37fe00c2deff28e44593e330aa6191f4
